chart/nats-connector/templates/deployment.yml

```diff
@@ -0,0 +1,78 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ template "connector.name" . }}
+    component: nats-connector
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    # Helm/k8s recommended label names v
+    app.kubernetes.io/name: {{ template "connector.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.Version }}
+    app.kubernetes.io/component: nats-connector
+    app.kubernetes.io/part-of: openfaas
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+  name: {{ template "connector.fullname" .}}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "connector.name" . }}
+      component: nats-connector
+  
+  template:
+    metadata:
+      annotations:
+        prometheus.io.scrape: "false"
+      labels:
+        app: {{ template "connector.name" . }}
+        component: nats-connector
+    spec:
+      containers:
+        - name: nats-connector
+          image: {{ .Values.image }}
+          env:
+            {{- if .Values.upstream_timeout }}
+            - name: upstream_timeout
+              value: {{ .Values.upstream_timeout | quote }}
+            {{- end }}
+            - name: gateway_url
+              value: {{ .Values.gateway_url | quote }}
+            - name: topics
+              value: {{ .Values.topics | quote }}
+            - name: print_response
+              value: {{ .Values.print_response | quote }}
+            - name: print_response_body
+              value: {{ .Values.print_response_body | quote }}
+            {{- if .Values.basic_auth }}
+            - name: basic_auth
+              value: "true"
+            - name: secret_mount_path
+              value: "/var/secrets/"
+            {{- end }}
+            - name: topic_delimiter
+              value: {{ .Values.topic_delimiter | quote }}
+            - name: asynchronous_invocation
+              value: {{ .Values.asynchronous_invocation | quote }}
+            - name: broker_host
+              value: {{ .Values.broker_host | quote  }}
+            {{- if .Values.rebuild_interval }}
+            - name: rebuild_interval
+              value: {{ .Values.rebuild_interval | quote }}
+            {{- end }}
+          {{- if .Values.basic_auth }}
+          volumeMounts:
+            - name: auth
+              readOnly: true
+              mountPath: "/var/secrets/"
+          {{- end }}
+      {{- if .Values.basic_auth }}
+      volumes:
+        - name: auth
+          secret:
+            secretName: basic-auth
```

Comment:
I can't see any secret.yaml file which name is "basic-auth", should we need to add it?

Reply:
These secrets are created while OpenFaas deployment that's why there is no secret.yaml file.